.github/workflows/docbuild.yaml

```diff
@@ -0,0 +1,53 @@
+name: Documentation Build
+
+on:
+  push:
+    branches:
+      - master
+
+permissions:
+  contents: write
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v2
+        with:
+          path: matter
+          fetch-depth: 0
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: cache-pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-doc-pip
+      - name: Install packages
+        run: |
+          sudo apt update
+          DOXYGEN_VERSION=1.9.4
+          wget --no-verbose https://downloads.sourceforge.net/project/doxygen/rel-${DOXYGEN_VERSION}/doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz
+          tar xf doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz
+          echo "${PWD}/doxygen-${DOXYGEN_VERSION}/bin" >> $GITHUB_PATH
+      - name: Install base dependencies
+        working-directory: matter
+        run: |
+          sudo pip3 install -U pip
+          pip3 install -r docs/requirements.txt
+      - name: Build documentation
+        working-directory: matter/docs
+        run: |
+          mkdir -p _build/src
+          make html
+          touch _build/html/.nojekyll
+      - name: Deploy to gh-pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think we shouldn't push documentation to a branch of connectedhomeip repo as it will greatly increase the time to clone the repo. We had similar issues with doxygen in the past and solved that by using `external_repository` tag and pushing the doxygen output to https://github.com/project-chip/connectedhomeip-doc where we have github-pages set up (https://project-chip.github.io/connectedhomeip-doc/). If this job is going to produce documentation that includes the doxygen output we can probably completely get rid off `doxygen.yaml` and push output of this workflow to connectedhomeip-doc. Thoughts?

Reply:
Last deployment onto connectedhomeip-doc was Nov 8th 2021, so I don't think a lot of people miss that and we can reuse the target. I suppose the plan is to eventually include the entire doxygen, right?

Reply:
This PR provides the framework necessary for developers to expand upon and eventually include the entire doxygen yes. For now I will remove the doxygen part of the generated documentation and change the workflow to push it to `connectedhomeip-doc` instead, replacing the existing API documentation.